legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/sdlc/LegendDependencyManagement.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2024 Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.finos.legend.engine.ide.lsp.extension.sdlc;
+
+import java.nio.file.Path;
+import java.util.List;
+import org.finos.legend.engine.ide.lsp.extension.features.LegendVirtualFileSystemContentInitializer;
+import org.finos.legend.engine.language.pure.grammar.to.PureGrammarComposer;
+import org.finos.legend.engine.language.pure.grammar.to.PureGrammarComposerContext;
+import org.finos.legend.engine.shared.core.api.grammar.RenderStyle;
+import org.finos.legend.sdlc.protocol.pure.v1.PureModelContextDataBuilder;
+import org.finos.legend.sdlc.serialization.EntityLoader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class LegendDependencyManagement implements LegendVirtualFileSystemContentInitializer
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(LegendDependencyManagement.class);
+    private final List<LegendVirtualFile> dependenciesVirtualFiles;
+
+    public LegendDependencyManagement()
+    {
+        this.dependenciesVirtualFiles = loadDependenciesVirtualFiles();
+    }
+
+    private static List<LegendVirtualFile> loadDependenciesVirtualFiles()
+    {
+        LOGGER.info("Processing dependencies from classpath");
+        try (EntityLoader loader = EntityLoader.newEntityLoader(LegendDependencyManagement.class.getClassLoader()))
```

Comment:
Will this include the entities from the project itself, or just the dependencies?

Reply:
This is only from the dependencies, as the current project files are not on the classpath.